Build fails when versioning.enabled = false

**Describe the bug**
A clear and concise description of what the bug is.  Include the version(s) of DSpace where you've seen this problem. Link to examples if they are public.
DSpace 7.6
Building backend with disabled item versioning causes build to fail due to missing HandleIdentifierProvider in identifier-service.xml.

**To Reproduce**
Steps to reproduce the behavior:
1. set `versioning.enabled = false` in versioning.cfg
2. (optional) in identifier-service.xml comment out bean with id="org.dspace.identifier.HandleIdentifierProvider"
3. build backend

**Expected behavior**
The build should succeed.

**Related work**

PR is underway.
The failed tests look like this:
```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.607 s <<< FAILURE! - in org.dspace.xmlworkflow.XmlWorkflowFactoryTest
[ERROR] org.dspace.xmlworkflow.XmlWorkflowFactoryTest  Time elapsed: 0.607 s  <<< ERROR!
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.dspace.app.sherpa.submit.SHERPASubmitService' defined in file [F:\workspace\dspace7\src\dspace\dspace-api\target\testing\dspace\config\spring\api\sherpa.xml]: Cannot resolve reference to bean 'org.dspace.app.sherpa.submit.SHERPASubmitConfigurationService' while setting bean property 'configuration'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.dspace.app.sherpa.submit.SHERPASubmitConfigurationService' defined in file [F:\workspace\dspace7\src\dspace\dspace-api\target\testing\dspace\config\spring\api\sherpa.xml]: Cannot create inner bean 'org.dspace.app.sherpa.submit.MetadataValueISSNExtractor#743997e0' of type [org.dspace.app.sherpa.submit.MetadataValueISSNExtractor] while setting bean property 'issnItemExtractors' with key [0]; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating 
bean with name 'org.dspace.app.sherpa.submit.MetadataValueISSNExtractor#743997e0': Unsatisfied dependency expressed through field 'itemService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.dspace.content.ItemServiceImpl#0': Unsatisfied dependency expressed through field 'handleService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.dspace.handle.HandleServiceImpl#0': Unsatisfied dependency expressed through field 'siteService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.dspace.content.SiteServiceImpl#0': Unsatisfied dependency expressed through field 'metadataValueService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.dspace.content.MetadataValueServiceImpl#0': Unsatisfied dependency 
expressed through field 'authorizeService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.dspace.authorize.AuthorizeServiceImpl#0': Unsatisfied dependency expressed through field 'bitstreamService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.dspace.content.BitstreamServiceImpl#0': Unsatisfied dependency expressed through field 'relationshipService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.dspace.content.RelationshipServiceImpl#0': Unsatisfied dependency expressed through field 'entityTypeService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.dspace.content.EntityTypeServiceImpl#0': Unsatisfied dependency expressed through field 'groupService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.dspace.eperson.GroupServiceImpl#0': Unsatisfied dependency expressed through field 'collectionService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.dspace.content.CollectionServiceImpl#0': Unsatisfied dependency expressed through field 'resourcePolicyService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.dspace.authorize.ResourcePolicyServiceImpl#0': Unsatisfied dependency expressed through field 'contentServiceFactory'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'contentServiceFactory': Unsatisfied dependency expressed through field 'dSpaceObjectServices'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.dspace.content.CommunityServiceImpl#0': Unsatisfied dependency expressed through field 'identifierService'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.dspace.identifier.service.IdentifierService' defined in file [F:\workspace\dspace7\src\dspace\dspace-api\target\testing\dspace\config\spring\api\identifier-service.xml]: Cannot resolve reference to bean 'org.dspace.identifier.HandleIdentifierProvider' while setting bean property 'providers' with key [0]; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.dspace.identifier.HandleIdentifierProvider' defined in file [...\src\dspace\dspace-api\target\testing\dspace\config\spring\api\identifier-service.xml]: Invocation of init method failed; 

nested exception is java.lang.RuntimeException: the org.dspace.identifier.VersionedHandleIdentifierProvider is enabled, but the versioning is disabled.
```
Current test run result 
`[ERROR] Tests run: 337, Failures: 0, Errors: 76, Skipped: 1`

--- a/dspace/config/spring/api/identifier-service.xml
+++ b/dspace/config/spring/api/identifier-service.xml
@@ -4,13 +4,20 @@
        xsi:schemaLocation="http://www.springframework.org/schema/beans
            http://www.springframework.org/schema/beans/spring-beans-2.5.xsd">
 
-    <!-- Identifier Service Application Interface.  Will be autowired with
+    <!-- Identifier Service Application Interface. Will be autowired with
          any Identifier Providers present in Spring context.
     -->
     <bean id="org.dspace.identifier.service.IdentifierService"
           class="org.dspace.identifier.IdentifierServiceImpl"
           autowire="byType"
           scope="singleton"/>
+
+    <!-- If you disable versioning, you need to use the default HandleIdentifierProvider. -->
+    <!--
+    <bean id="org.dspace.identifier.HandleIdentifierProvider" class="org.dspace.identifier.HandleIdentifierProvider" scope="singleton">
+        <property name="configurationService" ref="org.dspace.services.ConfigurationService"/>
+    </bean>
+    -->
 
       <!-- If you enabled versioning, you should use one of the versioned
            handle identifier provider instead of the default one.
@@ -26,7 +33,7 @@
            a new version is created the previous version gets a new
            handle. This leads to a handle that points always to the
            newest version, but there is no permanent handle, that
-           will always keep pointing to the acutal newest one.
+           will always keep pointing to the actual newest one.
            -->
     <!--
     <bean id="org.dspace.identifier.HandleIdentifierProvider" class="org.dspace.identifier.VersionedHandleIdentifierProviderWithCanonicalHandles" scope="singleton">

